mk/genlink-config.mk

```diff
@@ -18,17 +18,61 @@
 ##
 
 ifeq ($(DEVICE),)
-$(error no DEVICE specified for linker script generator)
+$(warning no DEVICE specified for linker script generator)
```

Comment:
If DEVICE not being specified causes the build to fail there will be a good indication for the reason.
However it's possible that a Makefile includes genlink-config.mk while not needing it (yet). The above check is executed no matter what. That means that any and all make targets would fail because of it for no good reason.